SCons/Util/sctypes.py

```diff
@@ -10,11 +10,27 @@
 import os
 import pprint
 import re
-from typing import Optional
-
+from typing import Optional, TYPE_CHECKING
+from typing_extensions import TypeAlias
 from collections import UserDict, UserList, UserString, deque
 from collections.abc import MappingView, Iterable
 
+
```

Comment:
Let's make SCons.Utils.typing if this code is all about getting typing right, let's not add it to this type conversion file..

Reply:
or SCons.Utils.types ? @mwichmann thoughts?

Reply:
> or SCons.Utils.types ? @mwichmann thoughts?

not `types`, please. I've been down that road already (that's why the util sub-module became `sctypes`). I'm not sure if `typing` would have the same conflict problem, since there's a stdlib `typing` module.

Reply:
Then `sctyping` would make for a sensible workaround. If we'd rather avoid the conflict altogether, `hints` or `hinting` could be used instead

Reply:
`sctyping` would work for me, the others not a problem either.  The conflict issues that caused me to rename what I first created as `types` had to do with unit tests, not with SCons code proper, FWIW.

Reply:
sctyping works! Let's go with that.